### PR TITLE
Fix memory leaks

### LINKED ISSUE #58: Looks like a memory leak
```
==11349== 1,920 bytes in 4 blocks are still reachable in loss record 6,916 of 7,035
==11349==    at 0x4C297BD: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==11349==    by 0x54F9E44: Konsole::ColorScheme::setColorTableEntry(int, Konsole::ColorEntry const&) (in /usr/lib/libqtermwidget5.so.0.6.0)
==11349==    by 0x54FCB86: Konsole::ColorScheme::readColorEntry(QSettings*, int) (in /usr/lib/libqtermwidget5.so.0.6.0)
==11349==    by 0x54FDE5D: Konsole::ColorScheme::read(QString const&) (in /usr/lib/libqtermwidget5.so.0.6.0)
==11349==    by 0x54FE0C2: Konsole::ColorSchemeManager::loadColorScheme(QString const&) (in /usr/lib/libqtermwidget5.so.0.6.0)
==11349==    by 0x54FF4B8: Konsole::ColorSchemeManager::loadAllColorSchemes() (in /usr/lib/libqtermwidget5.so.0.6.0)
==11349==    by 0x54FFB7A: Konsole::ColorSchemeManager::allColorSchemes() (in /usr/lib/libqtermwidget5.so.0.6.0)
==11349==    by 0x5520279: QTermWidget::availableColorSchemes() (in /usr/lib/libqtermwidget5.so.0.6.0)
==11349==    by 0x55205DA: QTermWidget::setColorScheme(QString const&) (in /usr/lib/libqtermwidget5.so.0.6.0)
==11349==    by 0x437EFE: TermWidgetImpl::propertiesChanged() (in /home/frtb38/qterminal/qterminal)
==11349==    by 0x4379FB: TermWidgetImpl::TermWidgetImpl(QString const&, QString const&, QWidget*) (in /home/frtb38/qterminal/qterminal)
==11349==    by 0x438D94: TermWidget::TermWidget(QString const&, QString const&, QWidget*) (in /home/frtb38/qterminal/qterminal)
```

### PR DESCRIPTION
**Pull request by f2404**
Fixes #58

### CODE REVIEW

**Comment by f2404** [comment]
The second commit fixes (at least) the following memory leaks:
```
==3995== 1,048 bytes in 1 blocks are still reachable in loss record 6,851 of 7,024
==3995==    at 0x4C297BD: operator new[](unsigned long) (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==3995==    by 0x62FA8F3: QHashData::rehash(int) (in /usr/lib/libQt5Core.so.5.5.1)
==3995==    by 0x550DE9E: Konsole::KeyboardTranslator::addEntry(Konsole::KeyboardTranslator::Entry const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511C29: Konsole::KeyboardTranslatorManager::loadTranslator(QIODevice*, QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511F1E: Konsole::KeyboardTranslatorManager::loadTranslator(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551211F: Konsole::KeyboardTranslatorManager::findTranslator(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511F80: Konsole::KeyboardTranslatorManager::defaultTranslator() (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5512314: Konsole::KeyboardTranslatorManager::findTranslator(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x550091B: Konsole::Emulation::setKeyBindings(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551E066: TermWidgetImpl::createSession(QWidget*) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551E2C0: TermWidgetImpl::TermWidgetImpl(QWidget*) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551E336: QTermWidget::init(int) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)

==3995== 2,261 bytes in 75 blocks are still reachable in loss record 6,942 of 7,024
==3995==    at 0x4C28C10: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==3995==    by 0x62981F7: QArrayData::allocate(unsigned long, unsigned long, unsigned long, QFlags<QArrayData::AllocationOption>) (in /usr/lib/libQt5Core.so.5.5.1)
==3995==    by 0x629AA2F: QByteArray::reallocData(unsigned int, QFlags<QArrayData::AllocationOption>) (in /usr/lib/libQt5Core.so.5.5.1)
==3995==    by 0x629C509: QByteArray::remove(int, int) (in /usr/lib/libQt5Core.so.5.5.1)
==3995==    by 0x629C702: QByteArray::replace(int, int, char const*, int) (in /usr/lib/libQt5Core.so.5.5.1)
==3995==    by 0x550D1C5: Konsole::KeyboardTranslator::Entry::unescape(QByteArray const&) const (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5510A3A: Konsole::KeyboardTranslatorReader::readNext() (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5510EDC: Konsole::KeyboardTranslatorReader::nextEntry() (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511C1E: Konsole::KeyboardTranslatorManager::loadTranslator(QIODevice*, QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511F1E: Konsole::KeyboardTranslatorManager::loadTranslator(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551211F: Konsole::KeyboardTranslatorManager::findTranslator(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511F80: Konsole::KeyboardTranslatorManager::defaultTranslator() (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)

==3995== 4,224 bytes in 88 blocks are still reachable in loss record 6,982 of 7,024
==3995==    at 0x4C28C10: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==3995==    by 0x62FA208: QHashData::allocateNode(int) (in /usr/lib/libQt5Core.so.5.5.1)
==3995==    by 0x550DEE1: Konsole::KeyboardTranslator::addEntry(Konsole::KeyboardTranslator::Entry const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511C29: Konsole::KeyboardTranslatorManager::loadTranslator(QIODevice*, QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511F1E: Konsole::KeyboardTranslatorManager::loadTranslator(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551211F: Konsole::KeyboardTranslatorManager::findTranslator(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5511F80: Konsole::KeyboardTranslatorManager::defaultTranslator() (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x5512314: Konsole::KeyboardTranslatorManager::findTranslator(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x550091B: Konsole::Emulation::setKeyBindings(QString const&) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551E066: TermWidgetImpl::createSession(QWidget*) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551E2C0: TermWidgetImpl::TermWidgetImpl(QWidget*) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
==3995==    by 0x551E336: QTermWidget::init(int) (in /home/frtb38/qtermwidget/libqtermwidget5.so.0.6.0)
```

**Comment by f2404** [comment]
Anyone?

File: lib/qtermwidget.cpp [hunk 1]
```diff
@@ -306,6 +306,8 @@ void QTermWidget::init(int startnow)
 
 QTermWidget::~QTermWidget()
 {
+    delete ColorSchemeManager::instance();
```
Comment (philippwiesemann) [inline lib/qtermwidget.cpp]: This breaks the remaining `QTermWidget`s because they still use the singleton (which just got deleted).

Comment (f2404) [inline lib/qtermwidget.cpp]: Thanks... will look further into this.

Comment (pmattern) [inline lib/qtermwidget.cpp]: Any news on this?

Comment (f2404) [inline lib/qtermwidget.cpp]: Oops... thanks for reminding. I'll try to improve this within next couple of days.

**Comment by f2404** [comment]
I've revised my changes: the konsole code using Q_GLOBAL_STATIC has been taken - it removes the leaks.

**Pull request merged**